django/test/client.py

```diff
@@ -55,47 +55,57 @@ def __init__(self, message, last_response):
         self.redirect_chain = last_response.redirect_chain
 
 
-class FakePayload:
+class FakePayload(IOBase):
     """
     A wrapper around BytesIO that restricts what can be read since data from
     the network can't be sought and cannot be read outside of its content
     length. This makes sure that views can't do anything under the test client
     that wouldn't work in real life.
     """
 
-    def __init__(self, content=None):
+    def __init__(self, initial_bytes=None):
         self.__content = BytesIO()
         self.__len = 0
         self.read_started = False
-        if content is not None:
-            self.write(content)
+        if initial_bytes is not None:
+            self.write(initial_bytes)
 
     def __len__(self):
         return self.__len
 
-    def read(self, num_bytes=None):
+    def read(self, size=-1, /):
         if not self.read_started:
             self.__content.seek(0)
             self.read_started = True
-        if num_bytes is None:
-            num_bytes = self.__len or 0
+        if size == -1 or size is None:
+            size = self.__len
         assert (
-            self.__len >= num_bytes
+            self.__len >= size
         ), "Cannot read more than the available bytes from the HTTP incoming data."
-        content = self.__content.read(num_bytes)
-        self.__len -= num_bytes
+        content = self.__content.read(size)
+        self.__len -= len(content)
```

Comment:
I guess it's worked in the past, but I suppose that `.read()` on the underlying `self.__content` isn't strictly guaranteed to return `size` bytes - it may return less.

See the documentation of [`RawIOBase.read()`](https://docs.python.org/3/library/io.html#io.RawIOBase.read) which states:
> Fewer than size bytes may be returned if the operating system call returns fewer than size bytes.

So I'll leave this minor improvement in.